src/content/docs/contributing/subworkflows.md

```diff
@@ -377,6 +377,161 @@ If a new test dataset is added to [`tests/config/test_data.config`](https://gith
+
+```bash
+mkdir subworkflows/nf-core/<subworkflow>/tests
+```
+
+- Generate a test file from template for your subworkflow using nf-test
+
+```bash
+nf-test generate workflow subworkflows/nf-core/<subworkflow>/main.nf
+```
+
+:::note
+We use `workflow` instead of `process` as it is a subworkflow not a module.
+:::
+
+- Move the generated test file to the tests directory
+
+```bash
+mv subworkflows/nf-core/<subworkflow>/main.nf.test subworkflows/nf-core/<subworkflow>/tests/
+```
+
+- Open the `main.nf.test` file and change the path for the script to a relative path `../main.nf`
+
+```diff title="main.nf.test"
+nextflow_workflow {
+     name "Test Workflow SUBWORKFLOW"
+-    script "subworkflows/nf-core/<subworkflow>/main.nf"
++    script "../main.nf"
+     process "SUBWORKFLOW"
+```
+
+- Then add tags to identify this module
+
+```groovy title="main.nf.test"
+tag "subworkflows"
+tag "subworkflows_nfcore"
+tag "subworkflows/<subworkflow>"
+tag "<subworkflow>"
+tag "<tool1>"
+tag "<tool1/sub-tool1>"
+tag "<tool2>"
+tag "<tool2/sub-tool2>"
+tag "<module2>"
+```
+
+:::note
+We require to have `tag subworkflows/<subworkflow>` as so it's picked up correctly.
+:::
+
+:::note
+Include the used tools and modules in the tags.
+:::
+
+- Provide a test name preferably indicating the test-data and file-format used. Example: `test("homo_sapiens - [bam, bai, bed] - fasta - fai")`
+
+:::note
+Multiple tests are allowed in a single test file.
+:::
+
+- If migrating an existing subworkflow, get the inputs from current pytest files `tests/subworkflow/nf-core/subworkflow/main.nf` and provide as positional inputs `input[0]` in nf-test file
+
+```groovy
+input[0] = [
+            [id:"ref"],
+            file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+           ]
+```
+
+- Next, in the `then` block we can write our assertions that are used to verify the test. A test can have multiple assertions but, we recommend enclosing all assertions in a `assertAll()` block as shown below:
+
+```groovy
+assertAll(
+            { assert workflow.success },
+            { assert snapshot(workflow.out).match() }
+          )
+```
+
+:::note
+It's `workflow.` and whereas with modules it's `process.`.
+:::
+
+- Run the test to create a snapshot of your module test. This will create a `.nf.test.snap` file
+
+```bash
+nf-test test --tag "<subworkflow>" --profile docker --update-snapshot
+```
+
+- Re-run the test again to verify if snapshots match
+
+```bash
+nf-test test --tag "<subworkflow>" --profile docker
+```
+
+- Create a new `tags.yml` in the `subworkflows/nf-core/<subworkflow>/tests/` folder and add only the corresponding subworkflow tag from `tests/config/pytest_modules.yml`
+
+```yaml
+subworkflows/<subworkflow>:
+  - subworkflows/nf-core/<subworkflow>/**
+```
+
+:::note
+Remove the corresponding tags from `tests/config/pytest_modules.yml` so that py-tests for the module will be skipped on github CI
```

Comment:
```suggestion
Remove the corresponding tags from `tests/config/pytest_modules.yml` so that py-tests for the subworkflow will be skipped on github CI
```